### PR TITLE
:seedling: E2E for clients/githubrepo/contributors.go

### PR DESCRIPTION
**Pull request by naveensrinivasan**
- Add an end-to-end test for `contributorsHandler`

#### What kind of change does this PR introduce?

(Is it a bug fix, feature, docs update, something else?)

- [ ] PR title follows the guidelines defined in our [pull request documentation](https://github.com/ossf/scorecard/blob/main/CONTRIBUTING.md#pr-process)

#### What is the new behavior (if this is a feature change)?**

- [ ] Tests for the changes have been added (for bug fixes/features)

#### Does this PR introduce a user-facing change?

For user-facing changes, please add a concise, human-readable release note to
the `release-note`

(In particular, describe what changes users might need to make in their
application as a result of this pull request.)

```release-note

```

### CODE REVIEW

File: clients/githubrepo/contributors_e2e_test.go [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+	BeforeEach(func() {
+		ctx := context.Background()
+		token := getGithubToken()
+		ts := oauth2.StaticTokenSource(
+			&oauth2.Token{AccessToken: token},
+		)
+
+		tc := oauth2.NewClient(ctx, ts)
+		client := github.NewClient(tc)
```
Comment (spencerschrock) [inline clients/githubrepo/contributors_e2e_test.go:39]: Any reason to do this over our `roundtripper.NewTransport` which the rest of the code uses?
 
 Like `clients/githubrepo/githubrepo_suite_test.go` (also in this PR)

File: clients/githubrepo/contributors_e2e_test.go [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+		contribHandler = &contributorsHandler{
+			ghClient: client,
+			ctx:      ctx,
+		}
+	})
+	Context("getContributors()", func() {
+		skipIfTokenIsNot(patTokenType, "GITHUB_TOKEN only")
```
Comment (spencerschrock) [inline clients/githubrepo/contributors_e2e_test.go:46]: one of these values is mismatched.

File: clients/githubrepo/githubrepo_suite_test.go [hunk 3]
```diff
@@ -55,6 +59,14 @@ func skipIfTokenIsNot(t tokenType, msg string) {
...
 }
 
+func getGithubToken() string {
+	token := os.Getenv("GITHUB_TOKEN")
+	if token == "" {
+		token = os.Getenv("GITHUB_AUTH_TOKEN")
+	}
+	return token
+}
+
```
Comment (spencerschrock) [inline clients/githubrepo/githubrepo_suite_test.go:69]: not sure this is needed if reworking to use roundtripper.NewTransport

**Review by spencerschrock — APPROVED** [review]

File: clients/githubrepo/contributors_e2e_test.go (hunk 1 above)
Comment (naveensrinivasan) [reply clients/githubrepo/contributors_e2e_test.go]: Agreed, I took care of it.

File: clients/githubrepo/githubrepo_suite_test.go (hunk 3 above)
Comment (naveensrinivasan) [reply clients/githubrepo/githubrepo_suite_test.go]: Agreed, I took care of it.

File: clients/githubrepo/contributors_e2e_test.go (hunk 2 above)
Comment (naveensrinivasan) [reply clients/githubrepo/contributors_e2e_test.go:46]: Sorry, i don't understand.

Comment (spencerschrock) [reply clients/githubrepo/contributors_e2e_test.go:46]: `skipIfTokenIsNot(patTokenType...` means the test only runs if run with a PAT.
But the skip message says "GITHUB_TOKEN only" implying that it should only run with a github token.

So it should either be:
```
skipIfTokenIsNot(patTokenType, "PAT only")
```
or
```
skipIfTokenIsNot(githubWorkflowDefaultTokenType, "GITHUB_TOKEN only")
```

Comment (naveensrinivasan) [reply clients/githubrepo/contributors_e2e_test.go]: 👍

**Pull request merged**